scripts/west_commands/runners/nrfjprog.py

```diff
@@ -76,22 +76,24 @@ def do_exec_op(self, op, force=False):
         elif op_type == 'program':
             cmd.append('--program')
             cmd.append(_op['firmware']['file'])
-            erase = _op['chip_erase_mode']
+            opts = _op['options']
+            erase = opts['chip_erase_mode']
             if erase == 'ERASE_ALL':
                 cmd.append('--chiperase')
-            elif erase == 'ERASE_PAGES':
-                cmd.append('--sectorerase')
-            elif erase == 'ERASE_PAGES_INCLUDING_UICR':
-                cmd.append('--sectoranduicrerase')
+            elif erase == 'ERASE_RANGES_TOUCHED_BY_FIRMWARE':
+                if self.family == 'nrf52':
```

Comment:
does nrf51/nrf53/nrf91 not need this too?

Reply:
That's a very good question. The code has been like this forever, but you may have uncovered a bug. Can you open an internal ticket so we can discuss it there? I can't copy the people that should be in the conversation here.